modules/sdk-coin-eth/src/lib/types.ts

```diff
@@ -112,11 +111,17 @@ export class EthTransactionData implements EthLikeTransactionData {
 
     if (this.tx.verifySignature()) {
       result.from = bufferToHex(this.tx.getSenderAddress().toBuffer());
-      result.r = bufferToHex(this.tx.r);
-      result.s = bufferToHex(this.tx.s);
+      // TODO: assert?
+      if (this.tx.r != undefined) {
+        result.r = bufferToHex(toUnsigned(this.tx.r));
```

Comment:
```suggestion
        result.r = bufferToHex(toUnsigned(this.tx.r as BN));
```
probably safe to remove the if condition and do something like this, incase r is undefined, current change will simply ignore which is not what we need, the above change should throw an error

Reply:
>  the above change should throw an error

I thought `as` is just a type-system thing, and no-op at runtime, so it will not throw an error.

Reply:
Yeah, but if the value passes indeed undefined, bufferToHex / toUnsigned from eth utils should throw an error right?, currently we are just skipping if this is undefined

Reply:
The previous code wasn't checking for it, and I'd assume this can never happen. `r` and `s` on signatures, is not something that can just not be there.

Reply:
Yup r and s should be there, the previous code would  throw an error in case when we passed in undefined and I think that should be the correct behaviour we would expect.

Reply:
Previous code as in the original code without the if condition before your change, if r is undefined, the current code just skips right without throwing an error?, When you pass in undefined to the util function from the eth dependency, it will trigger an error as far as I understand. Just wanted to keep the same behaviour of the code here.